### PR TITLE
feat(docs): FRMW-199 - Migrate rxStatusColumn docs

### PR DESCRIPTION
**Pull request by bry4n**
JIRA: https://jira.rax.io/browse/FRMW-199

* [x] Dev LGTM
* [x] Dev LGTM

### CODE REVIEW

File: src/rxStatusColumn/rxStatusColumn.js [hunk 1]
```diff
@@ -23,9 +24,51 @@ angular.module('encore.ui.rxStatusColumn', [])
...
+ * passed in for `status`. The `api` attribute will be explained below.
+ *
+ * We currently support six statuses, with corresponding CSS styles. Namely,
+ * `"ACTIVE"`, `"DISABLED"`, `"WARNING"`, `"ERROR"`, `"INFO"` and `"PENDING"`.
+ * If your code happens to already use those statuses, then you can simply pass
+ * them to the `status` attribute as appropriate. However, it's likely that
+ * internally you will be receiving a number of different statuses from your
+ * APIs, and will need to map them to these six statuses.
+ *
+ * The example in the demo shows a typical use of this directive, such as:
```
Comment (100stacks) [inline src/rxStatusColumn/rxStatusColumn.js]: When we break this documentation out of the the README, @CITguy has had me reference the component demo via **ngdocs** `@link`:
`{@link /#/components/rxStatusColumn demo}`

File: src/rxStatusColumn/rxStatusColumn.js [hunk 2]
```diff
@@ -88,9 +132,36 @@ angular.module('encore.ui.rxStatusColumn', [])
...
+ * `rx-status-header` attribute, i.e.
+ *
+ * <pre>
+ * <th rx-status-header></th>
+ * </pre>
+ * Note that status columns are sortable with
+ * {@link /#/components/rxSortableColumn rxSortableColumn}, just like any
+ * other column. The demo below shows an example of this.
+ *
+ * One few things to note about the demo: The `<th>` is defined as:
```
Comment (100stacks) [inline src/rxStatusColumn/rxStatusColumn.js]: Same as above for demo.

File: src/rxStatusColumn/rxStatusColumn.js [hunk 3]
```diff
@@ -103,8 +174,122 @@ angular.module('encore.ui.rxStatusColumn', [])
...
+ * });
+ * </pre>
+ *
+ * These mappings will be used throughout all instances of `rx-status-column`
+ * in your code.
+ *
+ * ## addAPI()
+ *
+ * Say that you are using three APIs in your product, `X`, `Y` and `Z`. Both
+ * `X` and `Y` define a status `"FOO"`, which you want to map to encore-ui's
```
Comment (100stacks) [inline src/rxStatusColumn/rxStatusColumn.js]: Suggest you change `encore-ui` to `EncoreUI`

File: src/rxStatusColumn/rxStatusColumn.js [hunk 4]
```diff
@@ -103,8 +174,122 @@ angular.module('encore.ui.rxStatusColumn', [])
...
+ * `"WARNING"`. You can declare this  mapping with
+ * `rxStatusMappings.addGlobal({ 'FOO': 'WARNING' })`. But your API `Z` also
+ * returns a `"FOO"` status, which you need mapped to encore-ui's
```
Comment (100stacks) [inline src/rxStatusColumn/rxStatusColumn.js]: Same suggestion...change `encore-ui` to `EncoreUI`

File: src/rxStatusColumn/rxStatusColumn.js [hunk 5]
```diff
@@ -103,8 +174,122 @@ angular.module('encore.ui.rxStatusColumn', [])
...
+ * `rxStatusMappings.addAPI()`, as follows:
+ *
+ * <pre>
+ * rxStatusMappings.addAPI('z', { 'FOO': 'ERROR' });
+ * </pre>
+ *
+ * Then in your template code, you would use `rx-status-column` as:
+ *
+ * <pre>
+ * <td rx-status-column status="{{ status }}" api="z">
```
Comment (100stacks) [inline src/rxStatusColumn/rxStatusColumn.js]: Missing closing `</td>` tag

**Comment by bry4n** [comment]
@100stacks thanks for feedback :+1:

File: src/rxStatusColumn/rxStatusColumn.js [hunk 6]
```diff
@@ -103,8 +176,122 @@ angular.module('encore.ui.rxStatusColumn', [])
...
+ *
+ * All six can also take `api` as a second, optional parameter. Thus we could
+ * define the `rxStatusMappings.addAPI({ 'FOO': 'ERROR' }, 'z')` example from
+ * above as:
+ *
+ * <pre>
+ * rxStatusMappings.mapToError('FOO', 'z');
+ * </pre>
+ *
+ * ## getInternalMapping()
```
Comment (CITguy) [inline src/rxStatusColumn/rxStatusColumn.js]: This section should be documented via a `@methodOf rxStatusColumn.service:rxStatusMappings` with appropriate `@param` docs.

File: src/rxStatusColumn/rxStatusColumn.js [hunk 7]
```diff
@@ -103,8 +176,122 @@ angular.module('encore.ui.rxStatusColumn', [])
...
+ *     'STANDBY': 'INFO',
+ *     'SUSPENDED': 'WARNING',
+ *     'FAILURE': 'ERROR'
+ * });
+ * </pre>
+ *
+ * These mappings will be used throughout all instances of `rx-status-column`
+ * in your code.
+ *
+ * ## addAPI()
```
Comment (CITguy) [inline src/rxStatusColumn/rxStatusColumn.js]: This section should be documented via a `@methodOf rxStatusColumn.service:rxStatusMappings` with appropriate `@param` docs.

File: src/rxStatusColumn/rxStatusColumn.js [hunk 8]
```diff
@@ -103,8 +176,122 @@ angular.module('encore.ui.rxStatusColumn', [])
...
  * A set of methods for creating mappings between a product's notion
- * of statuses, and the status identifiers used in encore-ui
+ * of statuses, and the status identifiers used in EncoreUI
+ *
+ * To accommodate different statuses, the `rxStatusMappings` factory includes
+ * methods for defining mappings from your own statuses to the six defined ones.
+ * The basic methods for this are `rxStatusMappings.addGlobal()` and
+ * `rxStatusMappings.addAPI()`.
+ *
+ * ## addGlobal()
```
Comment (CITguy) [inline src/rxStatusColumn/rxStatusColumn.js]: This section should be documented via a `@methodOf rxStatusColumn.service:rxStatusMappings` with appropriate `@param` docs.

File: src/rxStatusColumn/rxStatusColumn.js [hunk 9]
```diff
@@ -103,8 +176,122 @@ angular.module('encore.ui.rxStatusColumn', [])
...
+ * <td rx-status-column status="{{ status }}" api="z"></td>
+ * </pre>
+ *
+ * This will tell EncoreUI that it should first check if the passed in
+ * `status` was defined separately for an api `"z"`, and if so, to use that
+ * mapping. If `status` can't be found in the mappings defined for `"z"`,
+ * then it will fall back to the mappings you defined in your `.addGlobal()`
+ * call.
+ *
+ * ## mapToActive()/mapToWarning()/mapToError()/mapToInfo()/mapToPending()
```
Comment (CITguy) [inline src/rxStatusColumn/rxStatusColumn.js]: These methods should be documented via a `@methodOf rxStatusColumn.service:rxStatusMappings` with appropriate `@param` docs.

Comment (bry4n) [inline src/rxStatusColumn/rxStatusColumn.js]: updated.

**Comment by 100stacks** [comment]
LGTM :+1:

**Comment by CITguy** [comment]
LGTM

**Pull request merged**